### PR TITLE
HIVE-28497: Address CVE due to commons-codec:commons-codec:jar:1.11

### PR DESCRIPTION
**Pull request by KiranVelumuri**
### What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/HIVE-28497
Exclude commons-codec:commons-codec:jar:1.11 from org.apache.httpcomponents:httpclient

### Why are the changes needed?

To address vulnerability sonatype-2012-0050

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

Yes
Before exclusion: [before.txt](https://github.com/user-attachments/files/16848311/before.txt)
After exclusion: [after.txt](https://github.com/user-attachments/files/16848340/after.txt)

### How was this patch tested?

mvn dependency:tree -Dincludes=commons-codec:commons-codec:jar

### CODE REVIEW

File: standalone-metastore/metastore-server/pom.xml [hunk 1]
```diff
@@ -41,6 +41,12 @@
       <artifactId>httpclient</artifactId>
       <version>${httpcomponents.client.version}</version>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
```
[the remark this answers is not in the document]
Comment (KiranVelumuri) [reply standalone-metastore/metastore-server/pom.xml:45]: Yes, we're shipping commons-codec-1.11.jar as well. This is coming from standalone-metastore/metastore-server/target/apache-hive-standalone-metastore-server-4.1.0-SNAPSHOT-bin.tar.gz/apache-hive-metastore-4.1.0-SNAPSHOT-bin/lib.

**Comment by KiranVelumuri** [comment]
@zhangbutao Could you please review?

File: standalone-metastore/metastore-server/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhangbutao) [reply standalone-metastore/metastore-server/pom.xml:45]: I  built  `standalone-metastore` module by applied your patch, but found `commons-codec` jar was lost in apache-hive-metastore-4.1.0-SNAPSHOT-bin/lib.
Should we keep common-codec-1.15.jar in the lib?

[the remark this answers is not in the document]
Comment (KiranVelumuri) [reply standalone-metastore/metastore-server/pom.xml:45]: commons-codec is not declared as a direct dependency in this module and parent-module. Upon excluding commons-codec:1.11 version, non-vulnerable version of commons-codec is made available in the classpath from hadoop-common. So, I think we need not have commons-codec-1.15.jar as we are not declaring explicitly.

**Review by zhangbutao — APPROVED** [review]
+ LGTM

**Pull request merged**